docs/modules/release-notes/pages/5-4-0.adoc

```diff
@@ -0,0 +1,353 @@
+* Added the `stream()` method to the SQL service to return the stream of result rows.
+https://github.com/hazelcast/hazelcast/pull/24525[#24525]
+* Introduced a new configuration object to control the exact initial partition offsets when consuming records from Kafka via the Jet engine.
+https://github.com/hazelcast/hazelcast/pull/21546[#21546]
+* Added the new user code module (`UserCodeTransforms`) to run user codes in containerized environments. Currently, the module can be used with the Jet engine to create mappings with the user codes. #6464
+* Added a new UDP socket connector processor (`StreamUDPSocketP`) which can be used as a streaming source.
+It accepts a transform function to be applied to the body of the datagram packet before moving it to downstream. #6161
+* Aligned the behavior of `hashCode()` method of `KeyedWindowResult` with that of `Map.Entry`. #697
+* Boxing of partitionId is now avoided in the `getPartitionAssignment()` method when the partition pruning is not used. #486
+* Added the ability to limit the creation of objects through reflection. #296
+* Added the ability to use reusable processor meta supplier for `forceTotalParallelismOne ()` on random members without permissions. #227
+* Added a comparator for the High-Density Memory Store's index entries. Previously, on-heap entries comparators were used which causes incorrect query outputs. #52
+
+=== Connectors
+
+* Implemented a new SQL mapping option to define the preferred local parallelism for connectors that support this configuration. Currently, only the Kafka connector supports this.
+See xref:sql:mapping-to-kafka.adoc#creating-a-kafka-mapping[Creating a Kafka Mapping] for example configurations.
+https://github.com/hazelcast/hazelcast/pull/26194[#26194]
+* Removed the beta annotations from the MongoDB classes.
+https://github.com/hazelcast/hazelcast/pull/25743[#25743]
+* Added TLS support for MongoDB data connections.
+https://github.com/hazelcast/hazelcast/pull/25301[#25301]
+* Added Oracle database support to the JDBC SQL connector.
+https://github.com/hazelcast/hazelcast/pull/25255[#25255]
+* Added support for inline Avro schemas for Kafka mappings.
+https://github.com/hazelcast/hazelcast/pull/25207[#25207]
+* Added support for `DataSerializable` serialization to Mongo connector.
+https://github.com/hazelcast/hazelcast/pull/25197[#25197]
+* Check for existing resources for Mongo connector is now done only once; previously, it was performed on every processor.
+https://github.com/hazelcast/hazelcast/pull/24953[#24953]
+* Hazelcast JDBC connector now supports Microsoft SQL server.
+https://github.com/hazelcast/hazelcast/pull/25071[#25071]
+* Added the ability to configure the pool size of a MongoDB data connection. See xref:sql:mapping-to-mongo.adoc#creating-a-mongodb-mapping[Creating a MongoDB Mapping].
+https://github.com/hazelcast/hazelcast/pull/25027[#25027]
+
+=== Data Structures
+
+* Added support of specifying the maximum size of a CPMap.
```

Comment:
I think this and the following item can be skipped - they are just parts of introducing the CPMap data structure.